spec.html

```diff
@@ -44965,7 +44965,7 @@ <h1>Promise.race ( _iterable_ )</h1>
           1. If _result_ is an abrupt completion, then
             1. If _iteratorRecord_.[[Done]] is *false*, set _result_ to Completion(IteratorClose(_iteratorRecord_, _result_)).
             1. IfAbruptRejectPromise(_result_, _promiseCapability_).
-          1. Return ? _result_.
+          1. Return ! _result_.
```

Comment:
This only works if `_result_` is a normal completion, but IfAbruptRejectPromise in the step above unwraps the completion record.

Reply:
The call to the IfAbruptRejectPromise macro will cause the algorithm to exit, so this line isn't reachable in the "`_result_` is an abrupt completion" case.

Reply:
If the step before IfAbruptRejectPromise sets `_result_` to a normal completion, then IfAbruptRejectPromise will just unwrap that completion and thus this step is reachable.

Reply:
The step before IfAbruptRejectPromise cannot set `_result_` to a normal completion. When IteratorClose is passed an abrupt completion as its second argument, it always returns that completion.

Reply:
(These points were addressed in the commit message.)

Reply:
Can we replace the step above with this? It would have avoided my confusion:
```
1. If _iteratorRecord_.[[Done]] is *false*, perform Completion(IteratorClose(_iteratorRecord_, _result_)).
```

Reply:
I *think* that's equivalent, because (in these cases) if `_result_` is abrupt, then it's a throw completion, so `IteratorClose` never gets past step 5, so it always returns its second arg, so the `set _result_` is a no-op.

I'll leave it to the editors.

Reply:
Editors talked about this. Our preferred fix is a different version of IteratorClose for use when the second argument would be statically known to be a throw completion, along the lines of "IteratorCloseAndSuppressExceptions" or whatever. Then this could be

```
1. If _result_ is an abrupt completion, then
  1. If _iteratorRecord_.[[Done]] is *false*, perform IteratorCloseSuppressingExceptions(_iteratorRecord_).
  1. IfAbruptRejectPromise(_result_, _promiseCapability_).
```

which would be easier to follow.

Something to pursue in a separate PR, though.